src/main/java/org/jpeek/web/Reports.java

```diff
@@ -151,6 +128,34 @@
         return new TypedPages(new Pages(output));
     }
 
+    /**
+     * Extract classes from passed Jar file.
+     * @param path Jar file path
+     * @throws IOException If fails
+     */
+    private static void extractClasses(final Path path) throws IOException {
+        try (JarFile jar = new JarFile(path.toFile())) {
+            final Enumeration<JarEntry> entries = jar.entries();
+            while (entries.hasMoreElements()) {
+                final JarEntry entry = entries.nextElement();
+                final Path item = path.getParent().resolve(entry.getName());
+                if (entry.isDirectory()) {
+                    item.toFile().mkdir();
+                    continue;
+                }
+                if (!item.getParent().toFile().exists()) {
```

Comment:
@dgarus Please extract 'item.getParent().toFile()' to a variable, so that you do not need to repeat it.